### PR TITLE
Fix ty var parsing on class head

### LINKED ISSUE #4522: Parser accepts @ binders in class heads
## Description
The parser accepts `typeVarBinding` in a class head when it should be `typeVarBindingPlain`
https://github.com/purescript/purescript/blob/a915253de357239e62265953f364460bf22d0328/src/Language/PureScript/CST/Parser.y#L725C28-L725C42

## To Reproduce

This succeeds:
```purescript
class Foo @a
```

## Expected behavior
This should be a parse error, same as `data`, since the compiler implicitly introduces visible binders for class variables.

## Additional context
https://github.com/natefaubion/purescript-tidy/issues/114

## PureScript version

0.15.10

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Fixes #4522.

---

**Checklist:**

- [x] Added a file to CHANGELOG.d for this PR (see CHANGELOG.d/README.md)
- [ ] Added myself to CONTRIBUTORS.md (if this is my first contribution)
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
@f-f Just FYI. ARM CI builds aren't starting.

**Review by garyb — APPROVED** [review]

**Review by rhendric — APPROVED** [review]

File: CHANGELOG.d/bug_fix-class-head-ty-var.md
Comment (rhendric) [inline CHANGELOG.d/bug_fix-class-head-ty-var.md:1]: Rename to `fix_*.md`

**Comment by f-f** [comment]
@JordanMartinez ah yes, host was upgraded but I forgot to restart the runners, fixed now

File: CHANGELOG.d/bug_fix-class-head-ty-var.md
Comment (JordanMartinez) [reply CHANGELOG.d/bug_fix-class-head-ty-var.md:1]: Done. Thanks!

**Pull request merged**